master/buildbot/util/giturlparse.py

```diff
@@ -27,23 +27,39 @@
     r'(?P<proto>(https?://|ssh://|git://|))'
     r'((?P<user>.*)@)?'
     r'(?P<domain>[^\/:]+)(:((?P<port>[0-9]+)/)?|/)'
-    r'(?P<owner>.+)/(?P<repo>[^/]+?)(\.git)?$')
+    r'(?:(?P<path>.+))$')
```

Comment:
I think you can solve this problem with a more tricky regex:

wouldn't that work?
```
r'((?P<owner>.+)/)?(?P<repo>[^/]+?)(\.git)?$'
```

Reply:
Fixed now.